cli/client.go

```diff
@@ -380,6 +387,111 @@ func RobotsPartShellAction(c *cli.Context) error {
 	)
 }
 
+// checkUpdateResponse holds the values used to hold release information.
+type checkUpdateResponse struct {
+	Name       string `json:"name"`
+	TagName    string `json:"tag_name"`
+	TarballURL string `json:"tarball_url"`
+}
+
+// CheckUpdateAction is the corresponding Action for 'check-update'.
+func CheckUpdateAction(c *cli.Context) error {
+	if c.Bool(quietFlag) {
+		return nil
+	}
+
+	dateCompiledRaw := rconfig.DateCompiled
+	dateCompiled, err := time.Parse("2006-01-02", dateCompiledRaw)
+	if err != nil {
+		// user installed from go install or source
+		utils.UncheckedError(err)
+		return nil
+	}
+
+	conf, err := configFromCache()
+	if err != nil {
+		if !os.IsNotExist(err) {
+			utils.UncheckedError(err)
+			return nil
+		}
+		conf = &config{}
+	}
+
+	if conf.LastUpdateCheck == "" {
+		conf.LastUpdateCheck = time.Now().Format("2006-01-02")
+	  err := storeConfigToCache(conf)
+    if err != nil { 
+			utils.UncheckedError(err)
+    }
+	} else {
+		lastCheck, err := time.Parse("2006-01-02", conf.LastUpdateCheck)
+		if err != nil {
+			// corrupted date
+			utils.UncheckedError(err)
+			return nil
+		}
+		if time.Since(lastCheck) < time.Hour*24*7 {
+			return nil
+		}
+	}
```

Comment:
I would store the results of the check and print that (otherwise it'll be easy to miss it if the message only shows once every week), but invalidate the cache after 3 days